core/dispatcher/src/main/scala/whisk/core/container/ContainerPool.scala

```diff
@@ -175,19 +176,23 @@ class ContainerPool(
         info(this, s"Getting container for image $imageName with args " + args.mkString(" "))
         // Not a regular key. Doesn't matter in testing.
         val key = new ActionContainerId(s"instantiated." + imageName + args.mkString("_"))
-        getImpl(0, key, () => makeContainer(key, imageName, args)) map { _._1 }
+        getImpl(0, 0, key, () => makeContainer(key, imageName, args)) map { _._1 }
     }
 
     /**
      * Tries to get/create a container via the thunk by delegating to getOrMake.
      * This method will apply retry so that the caller is blocked until retry succeeds.
      */
     @tailrec
-    final def getImpl(position: Int, key: ActionContainerId, conMaker: () => FinalContainerResult)(implicit transid: TransactionId): Option[(Container, Option[RunResult])] = {
+    final def getImpl(tryCount: Int, position: Int, key: ActionContainerId, conMaker: () => FinalContainerResult)(implicit transid: TransactionId): Option[(Container, Option[RunResult])] = {
         val positionInLine = position - completedPosition.cur // this will be 1 if at the front of the line
         val available = slack()
         if (positionInLine > available) { // e.g. if there is 1 available, then I wait if I am second in line (positionInLine = 2)
             Thread.sleep(50) // TODO: replace with wait/notify but tricky to get right because of desire for maximal concurrency
+            if (tryCount == 200) {
+                warn(this, s"getImpl possibly stuck: position = $position   completed = ${completedPosition.cur}")
+                warn(this, s"getImpl possibly stuck: slack = $slack   maxActive = ${_maxActive}   activeCount = ${activeCount()}  startingCounter = ${startingCounter.cur}")
```

Comment:
```
-                warn(this, s"getImpl possibly stuck: position = $position   completed = ${completedPosition.cur}")
-                warn(this, s"getImpl possibly stuck: slack = $slack   maxActive = ${_maxActive}   activeCount = ${activeCount()}  startingCounter = ${startingCounter.cur}")
+                warn(this, s"""getImpl possibly stuck:
+                              | position = $position
+                              | completed = ${completedPosition.cur}")
+                              | slack = $available
+                              | maxActive = ${_maxActive}
+                              | activeCount = ${activeCount()}
+                              | startingCounter = ${startingCounter.cur}""".stripMargin)
```
since emit replaces `\n` with space this turns into a single line.